################################################################################
SHORT DESCRIPTION: 
################################################################################
Trace the TPM TCG vendor ID.

################################################################################
LONG DESCRIPTION: 
################################################################################
Output the TPM TCG vendor ID with the device ID and the revision ID to identify
the TPM device in use on the log output.

################################################################################
CHANGELOG 
################################################################################
Original author: Ross Philipson <philipsonr@ainfosec.com>
Port to 3.18: Eric Chanudet <chanudete@ainfosec.com>
Port to 4.14: Richard Turner <turnerr@ainfosec.com>
Port to 4.19: Richard Turner <turnerr@ainfosec.com>

################################################################################
REMOVAL 
################################################################################
This patch is for debugging purposes and can be safely removed.

################################################################################
UPSTREAM PLAN
################################################################################
None.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
None.

################################################################################
PATCHES 
################################################################################
--- a/drivers/char/tpm/tpm_tis_core.c
+++ b/drivers/char/tpm/tpm_tis_core.c
@@ -933,9 +933,9 @@ int tpm_tis_core_init(struct device *dev
 	if (rc < 0)
 		goto out_err;
 
-	dev_info(dev, "%s TPM (device-id 0x%X, rev-id %d)\n",
+	dev_info(dev, "%s TPM (vendor-id 0x%X device-id 0x%X, rev-id %d)\n",
 		 (chip->flags & TPM_CHIP_FLAG_TPM2) ? "2.0" : "1.2",
-		 vendor >> 16, rid);
+		 vendor & 0xffff, vendor >> 16, rid);
 
 	probe = probe_itpm(chip);
 	if (probe < 0) {
